cfme/infrastructure/virtual_machines.py

```diff
@@ -960,22 +1066,21 @@ def get_number_of_vms(do_not_navigate=False):
 @navigator.register(Template, 'All')
 @navigator.register(Vm, 'All')
 class VmAllWithTemplates(CFMENavigateStep):
-    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
-
-    def step(self, *args, **kwargs):
-        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', '/vm_infra/explorer')
-        accordion.tree('VMs & Templates', 'All VMs & Templates')
+    VIEW = VmsTemplatesAllView
+    prerequisite = NavigateToObject(Server, 'LoggedIn')
 
-    def resetter(self, *args, **kwargs):
-        reset_page()
+    def step(self):
+        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Virtual Machines')
+        self.view.sidebar.vmstemplates.tree.click_path('All VMs & Templates')
 
-    def am_i_here(self, *args, **kwargs):
-        return match_page(summary='All VMs & Templates')
+    def am_i_here(self):
+        return self.view.is_displayed and self.view.sidebar.vmstemplates.is_opened
 
 
 @navigator.register(Template, 'AllForProvider')
 @navigator.register(Vm, 'AllForProvider')
 class VmAllWithTemplatesForProvider(CFMENavigateStep):
+    VIEW = VmsTemplatesAllView
```

Comment:
I think you actually need a separate view here, with the check for sidebar selection in that view's is_displayed method.

Reason being - although `am_i_here()` covers the difference between the views during navigation, if a caller uses `view.is_displayed` apart from navigation, the results could be a false-positive.

cloud.instances has separate `InstanceAllView` and `InstanceProviderAllView` for this reason.